packages/core/src/Checkbox/Checkbox.tsx

```diff
@@ -76,21 +98,40 @@ const StyledInput = styled.input`
 `
 
 const Checkbox: React.FC<InferProps<typeof propTypes>> = React.forwardRef((props, ref) => {
-  // eslint-disable-next-line react/prop-types
-  const { disabled, size } = props
+  const inputRef = useRef()
+
+  useEffect(() => {
+    inputRef.current.indeterminate = props.indeterminate
+  }, [props.indeterminate])
 
+  // eslint-disable-next-line react/prop-types
+  const { disabled, size, indeterminate } = props
   // Add 4px to Icon's height and width to account for size reduction caused by adding padding to SVG element
   const borderAdjustedSize = size + 4
-
   return (
     <CheckBoxWrapper
       // eslint-disable-next-line react/prop-types
       theme={props.theme}
       color={props.color}
       disabled={disabled}
     >
-      <StyledInput type='checkbox' {...props} role='checkbox' ref={ref} />
+      <StyledInput
+        type='checkbox'
+        {...props}
+        role='checkbox'
+        aria-checked={props.indeterminate ? 'mixed' : props.checked}
```

Comment:
great accessibility addition here!

Reply:
You know I love helping blind people rent cars